engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```diff
@@ -769,7 +769,11 @@ public void processHeartbeatResponse(ContainerHeartbeatResponse rsp)
     }
 
     if (rsp.committedWindowId != lastCommittedWindowId) {
+
       lastCommittedWindowId = rsp.committedWindowId;
+
+      bufferServer.purge(lastCommittedWindowId);
```

Comment:
Working on that, here are the current finding.

1. Previous purge kept one extra window before the committed window. 
2. Couldn't find the history for that in GitHub. That change is present atleast from 2013.

Next steps

Will conduct few test scenario to see why we need a window before the committed window in the buffer server.